### PR TITLE
refactor: split up posthog.client

### PR DESCRIPTION
**Pull request by macobo**
## Problem

posthog/client.py has become really hard to work with. It also contains some 'speculative' code that isn't used.

Related: https://github.com/PostHog/product-internal/pull/408,

## Changes

- Split up posthog/client.py under posthog/clickhouse/client
- Keep posthog/client.py around for avoiding annoying merge conflicts
- Remove some unused code
- Simplify substitute_params code (replacing it with library implementation)
- Update a few imports

### CODE REVIEW

**Review by hazzadous — APPROVED** [review]

File: posthog/kafka_client/client.py [hunk 1]
```diff
@@ -216,7 +216,5 @@ def __init__(self):
     def produce(self, sql: str, topic: str, data: Dict[str, Any], sync: bool = True):
```
Comment (pauldambra) [inline posthog/kafka_client/client.py:216]: The `sync` parameter is unused here now. But some paths that call this still care about `sync`... should they not any more?

**Review by pauldambra — APPROVED** [review]
Only one bit that looks like a change not a refactor...

File: posthog/kafka_client/client.py (hunk 1 above)
Comment (macobo) [reply posthog/kafka_client/client.py:216]: async_execute always called sync_execute :shrug: 

We could remove the param, but given this only affects testing :shrug:

Comment (pauldambra) [reply posthog/kafka_client/client.py:216]: > async_execute always called sync_execute 🤷

🤣 

👍

**Pull request merged**